torchaudio/csrc/CMakeLists.txt

```
@@ -73,6 +73,10 @@ if(OpenMP_CXX_FOUND)
     )
 endif()
 
+if (BUILD_CUDA_CTC_DECODER)
+  add_subdirectory(cuctc)
+endif()
```

Can we move this to the end of `CMakeLists.txt` at the root level?
That way, this CMakeLists is focused on the configuration of libtorchaudio.

https://github.com/pytorch/audio/blob/5472cdae84cf1ed3f098be196df06d791c301d43/CMakeLists.txt#L150-L157

---

Also please add additional guard of `USE_CUDA`.

```suggestion
if (BUILD_CUDA_CTC_DECODER)
  if (NOT USE_CUDA)
    message(FATAL "BUILD_CUDA_CTC_DECODER=1 but USE_CUDA=0.")
  endif()
  add_subdirectory(torchaudio/csrc/cuctc)
endif()
```